source/kraken/realtime.cpp

```diff
@@ -528,31 +564,21 @@ void handle_realtime(const std::string& id,
     auto log = log4cplus::Logger::getInstance("realtime");
     LOG4CPLUS_TRACE(log, "realtime trip update received");
 
-    if (! is_handleable(trip_update)
+    if (! is_handleable(trip_update, *data.pt_data, is_realtime_add_enabled, is_realtime_add_trip_enabled)
             || ! check_trip_update(trip_update)) {
         LOG4CPLUS_DEBUG(log, "unhandled real time message");
         return;
     }
-    // if is_realtime_add_enabled = false, this disables the adding trip option too
-    if ((! is_realtime_add_enabled && is_added_stop_time(trip_update)) ||
-        (! is_realtime_add_enabled && is_added_trip(trip_update))) {
-        LOG4CPLUS_DEBUG(log, "Add stop time is disabled: ignoring trip update id "
-                << trip_update.trip().trip_id()
-                << " with ADDED/ADDED_FOR_DETOUR stop times or ADDITIONAL_SERVICE effect");
-        return;
-    }
-    if (! is_realtime_add_trip_enabled && is_added_trip(trip_update)) {
-        LOG4CPLUS_DEBUG(log, "Add trip is disabled: ignoring trip update id "
-                << trip_update.trip().trip_id()
-                << " with ADDITIONAL_SERVICE effect");
-        return;
-    }
 
-    auto meta_vj = data.pt_data->meta_vjs.get_mut(trip_update.trip().trip_id());
-    if (! meta_vj) {
-        LOG4CPLUS_ERROR(log, "unknown vehicle journey " << trip_update.trip().trip_id());
-        // TODO for trip().ADDED, we'll need to create a new VJ
-        return;
+    // Get or create meta VJ
+    navitia::type::MetaVehicleJourney* meta_vj = nullptr;
+    if (! data.pt_data->meta_vjs.exists(trip_update.trip().trip_id())) {
```

Comment:
We could save a search and still use get_mut (test if empty).

Reply:
for me, it just a reading logic. It is more clear to have exists and 2 method for read (get) and create (create). 
When you don't know the code this is more complex to understand with only _get_mut_... that was my case.

On the other hand, if you want to take into account your remark on `We add it only if it's an ADDED trip, I think.
Otherwise it should be rejected.`

You need to have something like that
```
bool exists = data.pt_data->meta_vjs.exists(trip_update.trip().trip_id()
if (!exist && new trip = added){
   Create meta VJ with emplace
}
else if (!exist && new trip != added) {
   // Don't want to create a meta VJ
   return
}
else {
    // meta vj exist
   get_mut() to retrieve it
}
```